filebeat/prospector/prospector_log.go

```diff
@@ -56,6 +56,21 @@ func (p *ProspectorLog) Run() {
 		p.Prospector.states.Cleanup()
 		logp.Debug("prospector", "Prospector states cleaned up.")
 	}
+
+	// Cleanup of removed files will only happen after next scan. Otherwise it can happen that not all states
+	// were updated before cleanup is called
+	if p.config.CleanRemoved {
+		for _, state := range p.Prospector.states.GetStates() {
+			_, err := os.Stat(state.Source)
+			if err != nil {
+				state.TTL = 0
```

Comment:
@urso During implementation I realised, that the above `Stat(path)` command does not return an error, when the file is not readable (which makes kind of sense). An error is only returned, if we try to open the file.

As I suggest to remove the file anyways only if it is really removed, I think the current implemented behaviour is as expected.